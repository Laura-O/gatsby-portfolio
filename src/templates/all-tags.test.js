import React from 'react';
import { shallow } from 'enzyme';

import AllTags from './all-tags';

describe('blog', () => {
  test('two tags', () => {
    const renderProps = {
      pathContext: {
        tags: ['tag1', 'tag2'],
      },
    };
    const wrapper = shallow(<AllTags {...renderProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('no tags', () => {
    const renderProps = {
      pathContext: {},
    };
    const wrapper = shallow(<AllTags {...renderProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
