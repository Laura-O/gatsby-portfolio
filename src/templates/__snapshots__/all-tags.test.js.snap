// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog no tags 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AllTags
    pageContext={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`blog two tags 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AllTags
    pageContext={
      Object {
        "tags": Array [
          "tag1",
          "tag2",
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <li>
          <UNDEFINED
            to="/tags/tag1"
          >
            tag1
          </UNDEFINED>
        </li>
        <li>
          <UNDEFINED
            to="/tags/tag2"
          >
            tag2
          </UNDEFINED>
        </li>
      </ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <UNDEFINED
              to="/tags/tag1"
            >
              tag1
            </UNDEFINED>
          </li>,
          <li>
            <UNDEFINED
              to="/tags/tag2"
            >
              tag2
            </UNDEFINED>
          </li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "tag1",
          "nodeType": "host",
          "props": Object {
            "children": <UNDEFINED
              to="/tags/tag1"
            >
              tag1
            </UNDEFINED>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "tag1",
              "to": "/tags/tag1",
            },
            "ref": null,
            "rendered": "tag1",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "tag2",
          "nodeType": "host",
          "props": Object {
            "children": <UNDEFINED
              to="/tags/tag2"
            >
              tag2
            </UNDEFINED>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "tag2",
              "to": "/tags/tag2",
            },
            "ref": null,
            "rendered": "tag2",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <li>
            <UNDEFINED
              to="/tags/tag1"
            >
              tag1
            </UNDEFINED>
          </li>
          <li>
            <UNDEFINED
              to="/tags/tag2"
            >
              tag2
            </UNDEFINED>
          </li>
        </ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <UNDEFINED
                to="/tags/tag1"
              >
                tag1
              </UNDEFINED>
            </li>,
            <li>
              <UNDEFINED
                to="/tags/tag2"
              >
                tag2
              </UNDEFINED>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "tag1",
            "nodeType": "host",
            "props": Object {
              "children": <UNDEFINED
                to="/tags/tag1"
              >
                tag1
              </UNDEFINED>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "tag1",
                "to": "/tags/tag1",
              },
              "ref": null,
              "rendered": "tag1",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "tag2",
            "nodeType": "host",
            "props": Object {
              "children": <UNDEFINED
                to="/tags/tag2"
              >
                tag2
              </UNDEFINED>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "tag2",
                "to": "/tags/tag2",
              },
              "ref": null,
              "rendered": "tag2",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
